internal/controller/imagerepository_controller_test.go

```diff
@@ -277,66 +279,74 @@ func TestImageRepositoryReconciler_scan(t *testing.T) {
 	proxyAddr, proxyPort := test.NewProxy(t)
 
 	tests := []struct {
-		name           string
-		tags           []string
-		exclusionList  []string
-		annotation     string
-		db             *mockDatabase
-		proxyURL       *url.URL
-		wantErr        string
-		wantTags       []string
-		wantLatestTags []string
+		name          string
+		tags          []string
+		exclusionList []string
+		annotation    string
+		db            *mockDatabase
+		proxyURL      *url.URL
+		wantErr       string
+		wantChecksum  string
+		wantTags      []string
 	}{
 		{
 			name:    "no tags",
 			wantErr: "404 Not Found",
 		},
```

Comment:
@stefanprodan There's already a test for empty tags. The registry returns 404 (at least the one used in the tests). I tried adding a test like this and the behavior is the same:

```go
		{
			name:     "empty tags",
			tags:     []string{},
			db:       &mockDatabase{},
			wantTags: []string{},
		},
```